src/chrome/content/rules/xkcd.xml

```diff
@@ -88,15 +88,22 @@
 
         <exclusion pattern="^http://(www\.|m\.)?xkcd\.com/1663/" />
 
+		<test url="http://xkcd.com/1663/" />
+		<test url="http://m.xkcd.com/1663/" />
+		<test url="http://www.xkcd.com/1663/" />
 
-        <test url="http://xkcd.com/1663/" />
-        <test url="http://m.xkcd.com/1663/" />
-        <test url="http://www.xkcd.com/1663/" />
+	<exclusion pattern="^http://c\.xkcd\.com/" />
 
 
 	<rule from="^http://(www\.|m\.)?xkcd\.org/"
 		to="https://$1xkcd.com/" />
 
+	<rule from="^https://c\.xkcd\.com/"
+		to="http://c.xkcd.com/"
+		downgrade="1" />
+
+		<test url="https://c.xkcd.com/" />
+
```

Comment:
@Hainish Because http://xkcd.com/ is rewritten to https://xkcd.com/ and that page includes a link to `//c.xkcd.com/random/comic/`, which is effectively forced to https unless there's an active downgrade.

The certificate is valid, but the [certificate chain isn't complete](https://www.ssllabs.com/ssltest/analyze.html?d=c.xkcd.com) so it won't work on all clients. (It uses a popular CA, so it will probably work on almost all of them, though.)

SSL Labs shows that the server is also vulnerable to an old TLS MITM attack, so the value of https is somewhat diminished anyway.